lib/adb.js

```diff
@@ -1,1700 +1,59 @@
-  var cmd = 'ime set ' + imeId;
-  this.shell(cmd, cb);
-};
-
-ADB.prototype.hasInternetPermissionFromManifest = function (localApk, cb) {
-  this.checkAaptPresent(function (err) {
-    if (err) return cb(err);
-    logger.debug("Checking if has internet permission from manifest.");
-    prettyExec(this.binaries.aapt,
-               ['dump', 'badging', localApk],
-               { maxBuffer: 524288 }, function (err, stdout, stderr) {
-      if (err || stderr) {
-        logger.warn(stderr);
-        return cb(new Error("hasInternetPermissionFromManifest failed. " + err));
-      }
-      var hasInternetPermission = new RegExp("uses-permission:.*'android.permission.INTERNET'").test(stdout);
-      cb(null, hasInternetPermission);
-    });
-  }.bind(this));
-};
-
-ADB.prototype.reboot = function (cb) {
-  var adbCmd = "stop; sleep 2; setprop sys.boot_completed 0; start";
-  this.shell(adbCmd, function (err) {
-    if (err) return cb(err);
-    var bootCompleted = false;
-    var i = 90;
-    logger.debug('waiting for reboot, this takes time.');
-    async.until(
-      function test() { return bootCompleted; },
-      function fn(cb) {
-        i--;
-        if (i < 0) return cb(new Error('device didn\'t reboot within 90 seconds'));
-        if (i % 5 === 0) logger.debug('still waiting for reboot.');
-        this.shell("getprop sys.boot_completed", function (err, stdout) {
-          if (err) return cb(err);
-          bootCompleted = '1' === stdout.trim();
-          setTimeout(cb, 1000);
-        });
-      }.bind(this),
-      cb
-    );
-  }.bind(this));
-};
-
-ADB.getAdbServerPort = function () {
-  return process.env.ANDROID_ADB_SERVER_PORT || 5037;
-};
-
-module.exports = ADB;
+// transpile:mocha
+
+import _ from 'lodash';
+import path from 'path';
+import { SystemCallsHelper } from './helpers/system-calls.js';
+import methods from './tools/index.js';
+
+class ADB {
+
+ constructor (opts = {}) {
+   if (typeof opts.sdkRoot === "undefined") {
+     opts.sdkRoot = process.env.ANDROID_HOME || '';
+   }
+   this.sdkRoot = opts.sdkRoot;
+   this.udid = opts.udid;
+   this.appDeviceReadyTimeout = opts.appDeviceReadyTimeout;
+   this.useKeystore = opts.useKeystore;
+   this.keystorePath = opts.keystorePath;
+   this.keystorePassword = opts.keystorePassword;
+   this.keyAlias = opts.keyAlias;
+   this.keyPassword = opts.keyPassword;
+   this.adb = {path: "adb", defaultArgs:[]};
+   this.tmpDir = opts.tmpDir;
+   if (opts.remoteAdbHost) {
+     this.adb.defaultArgs.push("-H", opts.remoteAdbHost);
+   }
+   if (opts.remoteAdbPort) {
+     this.adb.defaultArgs.push("-P", opts.remoteAdbPort);
+   }
+   this.curDeviceId = null;
+   this.emulatorPort = null;
+   this.logcat = null;
+   this.binaries = {};
+   this.instrumentProc = null;
+   this.javaVersion = opts.javaVersion;
+   this.suppressKillServer = opts.suppressAdbKillServer;
+   this.sysCallsHelper = new SystemCallsHelper(this.sdkRoot);
+   this.jars = {};
```

Comment:
to keep this constructor from getting too long, i'd move the jar logic into its own function `initJars()` and simply call it from here